torch/csrc/autograd/variable.cpp

```diff
@@ -363,6 +364,16 @@ void bump_version(const Variable& self) {
   self.unsafeGetTensorImpl()->bump_version();
 }
 
+void forbid_in_autograd(const Variable& self) {
+  TORCH_CHECK(
```

Comment:
You need a bit more error checking here to ensure that:
- This is a leaf (no existing grad_fn)
- This is not already part of a graph (no grad_accumulator)